src/ocamlorg_frontend/layouts/package_layout.eml

```diff
@@ -60,14 +60,30 @@ Layout.render
             </li>
           </ol>
 
-          <div title="Sorry, in-package search is not yet implemented, but this is where it's going to appear." class="flex w-full items-center">
-            <input disabled type="search" name="q" class="focus:border-gray-800 text-gray-800 bg-gray-300 border-gray-300 h-10 rounded-l-md appearance-none px-4 flex-grow"
+          <form class="flex w-full items-center"
+            action="https://doc.sherlocode.com/api" method="GET"
+            @submit.stop.prevent="false"
+            >
+            <input id="in_package_search" type="search" name="q" class="focus:border-gray-800 text-gray-800 border-primary-600 h-10 rounded-l-md appearance-none px-4 flex-grow"
               autocomplete="off"
-              placeholder="Sorry, in-package search is not yet implemented, but this is where it's going to appear :-)">
-            <button disabled aria-label="search" class="h-10 rounded-r-md bg-gray-300 text-gray-800 flex items-center justify-center px-4">
+              placeholder="Search most recently-indexed version of this package..."
+              hx-get="https://doc.sherlocode.com/api"
+              hx-params="*"
+              hx-vals='js:{ "packages" : "<%s Dream.to_percent_encoded package.name %>", "limit" : 10, q: document.getElementById("in_package_search").value.trim()}'
+              hx-trigger="keyup[target.value.trim().length > 1] changed delay:300ms, search"
```

Comment:
Can we drop the 300ms delay? It feels painfully slow for a service that responds in 10-20ms :P

Reply:
Hm, I tried 200ms and it was already too fast for me: it would still send useless requests. :thinking:. Reducing to 250.

Reply:
From a UX perspective, it doesn't feel right if the user has the time to wonder if the search even works :P It's faaar above the threshold for the system to feel reactive. Are we really saving more than a few bytes from not sending queries asap?